### PR TITLE
Feature/PT-100

### PR DESCRIPTION
**Pull request by 100hel**

### CODE REVIEW

File: src/components/PrivateRoute.jsx [hunk 1]
```diff
@@ -1,12 +1,27 @@
...
+  function getComponent() {
+    if (!isLoggedIn) {
+      return () => <Redirect to={paths.LOGIN} />;
+    }
+
+    if (hasAccessToPath(path)) {
+      return Component;
+    }
+
+    return () => <Redirect to={paths.PROFILE} />
```
Comment (AdamSarach) [inline src/components/PrivateRoute.jsx:21]: Could you explain when and why we want to redirect user to `profile` view?

File: src/components/PropertyGrid.jsx [hunk 2]
```diff
@@ -9,7 +9,7 @@ const PropertyGrid = ({propertyGrid, updatePropertyGrid}) => {
     const entries = Object.entries(grid);
     for (const [key, value] of entries) {
       gridEntries.push(
-        <div className="propertyGrid-row">
+        <div className="propertyGrid-row" key={key}>
           <div className="propertyGrid-key">{key}</div>
           <div className="propertyGrid-value">
             {typeof value === 'boolean' || value === null ? JSON.stringify(value) : value}
```
Comment (AdamSarach) [inline src/components/PropertyGrid.jsx:15]: We also do not need a panel from line 53 and 56-63. It doesn't look complex - I think we can throw it away.

File: src/components/auth/Logout.jsx [hunk 3]
```diff
@@ -3,9 +3,12 @@ import {Redirect} from 'react-router-dom';
 import {useDispatch, useSelector} from 'react-redux';
 import {useTranslation} from 'react-i18next';
 
-import colors from '../../stylesheets/config/colors.module.scss';
```
Comment (AdamSarach) [inline src/components/auth/Logout.jsx:6]: I'm glad we use external/internal/style way with imports :)

File: src/redux/reducers/authReducer.js [hunk 4]
```diff
@@ -24,11 +25,43 @@ function setTokens(token, refreshToken) {
...
+      parsedToken.roles = role;
+    } else {
+      parsedToken.roles = [role];
+    }
+  }
+
+  return parsedToken;
+}
+
 const authReducer = createReducer(initialState, (builder) => {
```
Comment (AdamSarach) [inline src/redux/reducers/authReducer.js:58]: Just for info - React Toolkit has a nice feature called `createSlice` - we could consider using it in future

File: src/utilities/auth.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import store from '../redux/store';
+import {paths, roles} from './constants';
+
+const { ADMIN, SUPERVISOR, EDITOR } = roles;
```
Comment (AdamSarach) [inline src/utilities/auth.js:4]: We have also coordinator role 

https://github.com/technologiesforaccessibility/osmintegrator-wiki/wiki/Permissions-and-Roles#coordinator

File: src/components/PropertyGrid.jsx (hunk 2 above)
Comment (AdamSarach) [reply src/components/PropertyGrid.jsx:15]: Done in PT-142. No action needed here.

File: src/components/PrivateRoute.jsx (hunk 1 above)
Comment (100hel) [reply src/components/PrivateRoute.jsx:21]: This is the only view that all the Roles can access.

File: src/components/auth/Logout.jsx (hunk 3 above)
Comment (100hel) [reply src/components/auth/Logout.jsx:6]: Unfortunately nothing has changed in this regard. This line has been moved below the other imports.

File: src/redux/reducers/authReducer.js (hunk 4 above)
Comment (100hel) [reply src/redux/reducers/authReducer.js:58]: I discussed this topic with Przemek before the implementation and both approaches are correct. I just prefer `createReducer`.

**Comment by AdamSarach** [comment]
Conflicts

**Review by AdamSarach — APPROVED** [review]

**Pull request merged**